### ISSUE TITLE
`std.Uri.unescapeString` ignores malformed encoding with trailing '%'

### ISSUE THREAD

**Issue by gcoakes** [ziglang/zig#17834]
### Zig Version

0.12.0-dev.1369+a09ba455c

### Steps to Reproduce and Observed Behavior

The following test passes:

```zig
const std = @import("std");

test {
    const decoded = try std.Uri.unescapeString(std.testing.allocator, "/abc%");
    defer std.testing.allocator.free(decoded);
    try std.testing.expectEqualStrings("/abc", decoded);
}
```

### Expected Behavior

I don't think that test should pass, but I'm not entirely certain what the correct behavior should be. My cursory look didn't turn up anything in RFC 3986 suggesting corrective action when the percent-encoded string is malformed. The most pedantic would be to return an error. As a counter argument, `python3 -m http.server` serves a file named `abc%` from the file system when given a request with the target `/abc%`.

**Comment by linusg** [comment]
Looking at some other languages it appears there's no clear precedent for either erroring or preserving the percent sign:

**C - libcurl** (7.86)

```c
#include <curl/curl.h>

CURL *curl = curl_easy_init();
int decodelen;
char *decoded = curl_easy_unescape(curl, "/abc%", 5, &decodelen); // Returns "/abc%"
curl_free(decoded);
```

**Go** (1.21)

```go
import "net/url"

_, err := url.QueryUnescape("/abc%") // Returns err: `invalid URL escape "%"`
```

**JavaScript**

```js
decodeURI("/abc%") // Throws URIError
decodeURIComponent("/abc%") // Throws URIError
```

**PHP** (8.2)

```php
urldecode("/abc%"); // Returns "/abc%"
```

**Python** (3.12)

```python
import utllib.parse

urllib.parse.unquote("/abc%") # Returns "/abc%"
```

**Swift** (5.9)

```swift
import Foundation

"/abc%".removingPercentEncoding // Returns nil
```

**Comment by andrewrk** [comment]
Fixed in d8b5831dc4f0666dceef671f47d23b5447b25912. The function is documented like so:
```zig
/// Parses a URI string and unescapes all %XX where XX is a valid hex number. Otherwise, verbatim copies
/// them to the output.
```
That's a fine definition as any.
